libnetwork/drivers/bridge/setup_bridgenetfiltering.go

```diff
@@ -12,27 +12,44 @@ import (
 	"github.com/containerd/log"
 )
 
-// Enumeration type saying which versions of IP protocol to process.
-type ipVersion int
-
-const (
-	ipvnone ipVersion = iota
-	ipv4
-	ipv6
-	ipvboth
-)
+// setupIPv4BridgeNetFiltering checks whether IPv4 forwarding is enabled and, if
+// it is, sets kernel param "bridge-nf-call-iptables=1" so that packets traversing
+// the bridge are filtered.
+func setupIPv4BridgeNetFiltering(*networkConfiguration, *bridgeInterface) error {
+	if enabled, err := getKernelBoolParam("/proc/sys/net/ipv4/ip_forward"); err != nil {
+		log.G(context.TODO()).Warnf("failed to check IPv4 forwarding: %v", err)
+		return nil
+	} else if enabled {
+		return enableBridgeNetFiltering("/proc/sys/net/bridge/bridge-nf-call-iptables")
+	}
+	return nil
+}
 
-// getIPVersion gets the IP version in use ( [ipv4], [ipv6] or [ipv4 and ipv6] )
-func getIPVersion(config *networkConfiguration) ipVersion {
-	ipVer := ipv4
-	if config.AddressIPv6 != nil || config.EnableIPv6 {
-		ipVer |= ipv6
+// setupIPv6BridgeNetFiltering checks whether IPv6 forwarding is enabled for the
+// bridge and, if it is, sets kernel param "bridge-nf-call-ip6tables=1" so that
+// packets traversing the bridge are filtered.
+func setupIPv6BridgeNetFiltering(config *networkConfiguration, _ *bridgeInterface) error {
+	if !config.EnableIPv6 || config.AddressIPv6 == nil {
```

Comment:
I'd rather not check if `AddressIPv6` is nil, thus would make the code more future proof.

Reply:
Done.